File: versioned_docs/version-1.21.x/examples/go/src/buckets_rename.go
package main

import (
	"context"
	reduct "github.com/reductstore/reduct-go"
)

func main() {
	// Create a client and use the base URL and API token
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{
		APIToken: "my-token",
	})

	// Rename the bucket with the name "bucket-to-rename"
	bucket, err := client.GetBucket(context.Background(), "bucket-to-rename")
	if err != nil {
		panic(err)
	}

	err = bucket.Rename(context.Background(), "bucket-renamed")
	if err != nil {
		panic(err)
	}

	// Check that the bucket was renamed
	_, err = client.GetBucket(context.Background(), "bucket-renamed")
	if err != nil {
		panic(err)
	}
}

File: versioned_docs/version-1.21.x/examples/js/src/buckets_rename.mjs
import { Client } from "reduct-js";
import assert from "node:assert";

// Create a new client with the server URL and an API token
const client = new Client("http://127.0.0.1:8383", { apiToken: "my-token" });

// Rename the bucket 'bucket-to-rename' to 'bucket-renamed'
const bucket = await client.getBucket("bucket-to-rename");
await bucket.rename("bucket-renamed");

// Check that the bucket was renamed
assert(await client.getBucket("bucket-renamed"));
try {
  await client.getBucket("bucket-to-rename");
} catch (e) {
  assert(e.status === 404);
}
